app/services/potential_duplicate_finder_service.rb

```diff
@@ -18,7 +20,7 @@ def self.duplicates_finder
                 person_name.middle_name,person_address.address2 AS home_district,
```

Comment:
I think the code fetching patients is being repeated, what if the patients could be extracted into a helper method to avoid code duplication and this method can be called whenever it is needed.
```suggestion
                def self.fetch_patients
                      Person.joins(:names, :addresses)
                                 .joins(patient: :patient_programs)
                                .where(patient_program: { program_id: 33, voided: 0 })
                                .select('person.person_id, person.birthdate, person.gender, ...')
                end,
```

Reply:
I also saw that as well but the problem is the other is using find_each and the other is using in_batch can you suggest how envision how we can put them in a helper or you can modify it and update the PR with your suggestion.